lib/index.js

```diff
@@ -313,32 +306,29 @@ class Mcdev {
      */
     static async document(businessUnit, type) {
         Util.logger.info('mcdev:: Document');
+
+        const properties = await config.getProperties();
+        if (type && !MetadataTypeInfo[type]) {
+            Util.logger.error(`:: '${type}' is not a valid metadata type`);
+            return;
+        }
```

Comment:
should be done before getting the config to increase speed of execution.
also, given that this is highest level execution - should we not switch the error log into a throw error instead?